demisto_sdk/commands/validate/validate_manager.py

```diff
@@ -2896,3 +2902,132 @@ def check_and_validate_deprecated(
 
             return is_valid_as_deprecated
         return None
+
+    @error_codes("BA120,BA121,BA122,BA123")
+    def is_valid_path(self, path: Path) -> bool:
+        """
+        Args:
+            path (Path): path to the file
+        Returns:
+            bool: whether the file may be saved in this path:
+                - BA120: Under a folder that accepts first-level files (e.g. Widgets, but not Integrations)
+                - BA121: Under an allowed first-level folder (e.g. Integrations, not `integration` or `INTEGRATIONS`)
+                - BA122: Allowed file name under a pack root folder.
+                - BA123: Files are not allowed outside of a pack.
+        """
+        if path.is_dir():
+            raise ValueError(
+                "is_allowed_modified_file_path should not be run on folders"
+            )
+
+        def _handle_outside_pack() -> bool:
```

Comment:
try to have one function that formats the returned values and gets the Errors.func() as an argument.